sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaSource.java

```diff
@@ -333,134 +305,41 @@ public boolean advance() throws IOException {
     }
 
     protected boolean advanceToNext() throws IOException {
-      current = null;
-      // The content is reported first
-      if (metadataIterator == null) {
-        // Check if some content is already available
-        current = tikaHandler.getCurrent();
-
-        if (current == null && !Boolean.TRUE.equals(spec.getParseSynchronously())) {
-          long maxPollTime = 0;
-          long configuredMaxPollTime = spec.getQueueMaxPollTime() == null
-              ? TikaIO.Read.DEFAULT_QUEUE_MAX_POLL_TIME : spec.getQueueMaxPollTime();
-          long configuredPollTime = spec.getQueuePollTime() == null
-              ? TikaIO.Read.DEFAULT_QUEUE_POLL_TIME : spec.getQueuePollTime();
-
-          // Poll the queue till the next piece of data is available
-          while (current == null && maxPollTime < configuredMaxPollTime) {
-            boolean docEnded = tikaHandler.waitForNext(configuredPollTime);
-            current = tikaHandler.getCurrent();
-            // End of Document ?
-            if (docEnded) {
-              break;
-            }
-            maxPollTime += spec.getQueuePollTime();
-          }
-        }
-        // No more content ?
-        if (current == null && Boolean.TRUE.equals(spec.getReadOutputMetadata())) {
-          // Time to report the metadata
-          metadataIterator = Arrays.asList(tikaMetadata.names()).iterator();
-        }
-      }
-
-      if (metadataIterator != null && metadataIterator.hasNext()) {
-          String key = metadataIterator.next();
-          // The metadata name/value separator can be configured if needed
-          current = key + "=" + tikaMetadata.get(key);
+      if (!docParsed) {
```

Comment:
Just make start() say "current = ...; return true;" and make advance "return false;" and remove docParsed variable.

Reply:
Thanks for the review, sure, will deal with the above comments first